src/olympia/users/admin.py

```diff
@@ -130,6 +131,19 @@ def change_view(self, request, object_id, form_url='', extra_context=None):
         extra_context = extra_context or {}
         extra_context['has_users_edit_permission'] = acl.action_allowed(
             request, amo.permissions.USERS_EDIT)
+
+        lookup_field = UserProfile.get_lookup_field(object_id)
+        if lookup_field != 'pk':
+            try:
+                if lookup_field == 'email':
```

Comment:
I would remove this condition so that, if `lookup_field` is something else, it should raise `UserProfile.DoesNotExist`. Otherwise you'll have to add a `else` branch to handle the case where `lookup_field` is something else.

Reply:
Clarified via chat that this is not necessary. We should let the method of the parent class handle other `lookup_field`s.